power-sequencer/mihawk-cpld.cpp

```diff
@@ -676,64 +678,40 @@ void MIHAWKCPLD::clearCPLDregister()
     close(fd);
 }
 
-// Check for Mowgli's HDDFault
-int MIHAWKCPLD::checkFault(int reg)
+// Check for Mowgli's HDDFault status
+MIHAWKCPLD::HDDErrorCode MIHAWKCPLD::checkHDDError(int statusReg)
 {
-    int result = 0;
-    std::string i2cBus = "/dev/i2c-" + std::to_string(busId);
-
-    // open i2c device(CPLD-PSU-register table)
-    int fd = open(i2cBus.c_str(), O_RDWR | O_CLOEXEC);
-    if (fd < 0)
+    HDDErrorCode result;
+    if ((readFromCPLDErrorCode(statusReg)) & 1)
     {
-        std::cerr << "Unable to open i2c device \n";
+        return result = static_cast<HDDErrorCode>(1);
```

Comment:
You can just do:
```
return HDDErrorCode::_1;
```
here and in the other spots.   This would work though, so I'll leave it up to you if you want to change it.

Reply:
Thanks for the reminder, I have fixed this part :)